pysph/tools/interpolator.py

```diff
@@ -279,27 +394,45 @@ def _create_particle_array(self, x, y, z):
 
         hmax = self._get_max_h_in_arrays()
         h = hmax*np.ones_like(xr)
-        prop = np.zeros_like(xr)
         pa = get_particle_array(
             name='interpolate',
             x=xr, y=yr, z=zr, h=h,
-            number_density=np.zeros_like(xr),
-            prop=prop,
-            grad_x=np.zeros_like(xr),
-            grad_y=np.zeros_like(xr),
-            grad_z=np.zeros_like(xr)
+            number_density=np.zeros_like(xr)
         )
+        if not self.use_shepard:
+            pa.add_property('mommat', stride=16)
+            pa.add_property('p_sph', stride=4)
+            pa.add_property('prop', stride=4)
+        else:
+            pa.add_property('prop')
+
         return pa
 
     def _compile_acceleration_eval(self, arrays):
         names = [x.name for x in self.particle_arrays]
         if self.equations is None:
             if self.use_shepard:
-                equations = [InterpolateFunction(dest='interpolate',
-                                                 sources=names)]
+                equations = [
+                    InterpolateFunction(dest='interpolate',
+                                        sources=names)
+                ]
             else:
```

Comment:
So here the original SPH interpolation is missing, this can be retained.

Reply:
done